File: pinned/lib/models/chatbot_model.js
// Import mongoose
const mongoose = require("mongoose");
const { Schema } = mongoose;

// Define chatbot schema
const chatbotDBSchema = new Schema(
  {
    email: {
        type: String,
        required: true
      },
      conversations: [{
        sender: String,
        message: String,
        createdAt: { type: Date, default: Date.now }
      }]
  }
);

// 모델 생성
const chatbotDBModel = mongoose.model("chatbotDB", chatbotDBSchema);

// 모델 내보내기
module.exports = chatbotDBModel;

File: pinned/lib/models/chatbot_model.dart
import 'package:flutter/material.dart';

enum Sender { user, ai }

class ChatbotModel {
  final int id;
  final String email;
  final Enum sender;
  final String text;
  final DateTime createdAt;

  ChatbotModel(
      {required this.id,
      required this.email,
      required this.sender,
      required this.text,
      required this.createdAt});
}
